src/radical/pilot/agent/launch_method/aprun.py

```diff
@@ -1,221 +1,84 @@
 
-__copyright__ = "Copyright 2016, http://radical.rutgers.edu"
-__license__   = "MIT"
+__copyright__ = 'Copyright 2016-2021, The RADICAL-Cybertools Team'
+__license__   = 'MIT'
 
+import os
 
-import pprint
 import radical.utils as ru
 
 from .base import LaunchMethod
 
-# maximum length of command line arguments we support. Beyond that we use
-# a hostfile to specify the process layout.
-ARG_MAX = 4096
-
 
 # ------------------------------------------------------------------------------
 #
 class APRun(LaunchMethod):
 
     # --------------------------------------------------------------------------
     #
-    def __init__(self, name, cfg, session):
-
-        LaunchMethod.__init__(self, name, cfg, session)
+    def _configure(self):
 
+        self.launch_command = ru.which('aprun')
+        # only one instance of `aprun` is allowed per node at any time,
+        # otherwise multiple instances will be executed sequentially
```

Comment:
Do you know if this universally true for aprun, or is that  a system config setting on Theta?